extensions/qute/deployment/src/main/java/io/quarkus/qute/deployment/Types.java

```diff
@@ -133,10 +133,13 @@ static boolean containsTypeVariable(Type type) {
     }
 
     static boolean isAssignableFrom(Type type1, Type type2, IndexView index, Map<DotName, AssignableInfo> assignableCache) {
-        // TODO consider type params in assignability rules
-        if (type1.kind() == Kind.ARRAY && type2.kind() == Kind.ARRAY) {
-            return isAssignableFrom(type1.asArrayType().component(), type2.asArrayType().component(), index, assignableCache);
+        if (type1.kind() == Kind.ARRAY) {
```

Comment:
What if `type2` is an array and `type1` isn't?

Reply:
Then the method returns `false` ... see https://github.com/quarkusio/quarkus/pull/31112/files#diff-9fdf5c1c1f88f93d3b6b0d4e5f0ebb70b73d7309644b5a036ca9d012123f06e0R49.

Reply:
OK :)